src/value.mts

```diff
@@ -848,6 +866,8 @@ export class ObjectValue extends Value implements ObjectInternalMethods<ObjectVa
   static {
     Object.defineProperty(this.prototype, 'type', { value: 'Object' });
   }
+
+  declare static [Symbol.hasInstance]: (value: unknown) => value is ObjectValue;
```

Comment:
This should be the default behaviour of `instanceof`, we don't need this signature to specify it, so I'm wondering if we can remove the whole `hasInstance` or not.

Reply:
This is needed to override the `(value: unknown) => value is Value` declaration in `BaseValue`, otherwise `value instanceof ObjectValue` will break with the fixed `Symbol.hasInstance` method type in `BaseValue`.

Reply:
I see, it is because how we define Value